_posts/2020-02-03-transparent-logs.md

```diff
@@ -70,12 +70,18 @@ In particular, the Datadog TUF and in-toto integration mentioned in the last col
 
-We would like to thank Justin Cappos, Nick Coghlan, Lois Anne DeLong, Ernest W. Durbin III, Sumana Harihareswara, Joshua Lock, Santiago Torres-Arias, and the Python community for their feedback.
+We would like to thank Justin Cappos, Nick Coghlan, Lois Anne DeLong, Ernest W. Durbin III, Sumana Harihareswara, Joshua Lock, Santiago Torres-Arias, Filippo Valsorda, and the Python community for their feedback.
 
 ## Footnotes
 
-[^1]: In fact, assuming that the TL and package repository are independent, this does not even require attackers to compromise the TL. Since the TL would [automatically](https://go.googlesource.com/proposal/+/master/design/25530-sumdb.md#checksum-database) fetch missing versions of packages, all attackers would have to to do is to add malicious versions of packages to the repository (such as the GitHub repository belonging to the package developers), and somehow convince developers to refer to these malicious versions (say, by publishing new tags on GitHub). In this sense, TLs still depend on [Trust-On-First-Use (TOFU](https://go.googlesource.com/proposal/+/master/design/25530-sumdb.md#module-authentication-with)). While Go encourages pinning packages using [Semantic Versioning](https://semver.org/), which ameliorates the issue to some extent, the fact remains that malicious versions of packages can still be added automatically, which is especially problematic when package managers such as [`pip`](https://pypi.org/project/pip/) automatically try to find the latest versions of packages.
+[^1]: Since this article was published, we have [learned](https://github.com/secure-systems-lab/ssl-site/issues/95) that the procedure for recovering from a key compromise for TLs is, to _some_ extent, comparable to PEP 458 security model for TUF. It is best illustrated using the following scenario. Suppose that before time N, online key P was used. At time N, online key P is compromised. After time N, we switch trust to online key Q instead. Using TLs (e.g., Go sumdb), the log will be signed using both online keys P and Q. This is so that [Go binary distributions](https://golang.org/doc/install) released before time N that are effectively baked with P can continue to trust the TL. Subsequent Go binary distributions will be baked with Q instead. Presumably, P will be considered deprecated / compromised, and some grace period will be allowed before P is completely revoked (i.e., no longer used to sign the TL).  Using TUF (PEP 458), there is no need to issue a new software update to permanently replace trust in P with Q. The TUF metadata repository administrators would use the offline keys for the `targets` role (not even necessarily the higher-level `root` role) to do so. Both old and new versions of the package manager can continue to permanently switch trust from P to Q, despite [backwards-incompatible](https://github.com/theupdateframework/taps/pull/107) changes to TUF metadata. We feel that this is a subtle but important difference.
```

Comment:
It might be good to note that by rotating trust from P to Q in TUF happens automatically during the next update cycle, so there is no wait time before P is effectively deprecated.